app/views/contexts/calculated_summary.py

```diff
@@ -0,0 +1,188 @@
+from app.jinja_filters import (
+    get_formatted_currency,
+    format_number,
+    format_unit,
+    format_percentage,
+)
+from app.questionnaire.schema_utils import (
+    choose_question_to_display,
+    get_answer_ids_in_block,
+)
+from app.questionnaire.path_finder import PathFinder
+from app.questionnaire.location import Location
+from app.views.contexts.summary.group import Group
+
+
+def build_groups_for_section(answer_store, list_store, metadata, schema, section):
+    path_finder = PathFinder(schema, answer_store, metadata, list_store=list_store)
+
+    section_path = path_finder.routing_path(section['id'])
+
+    location = Location(section['id'])
```

Comment:
Wouldn't work for repeats?.

Reply:
Nope. I've moved calculated summary out of the summary, but I think it's out of scope for this PR.